HikVisionSource/build.gradle

```diff
@@ -0,0 +1,79 @@
+group 'io.vantiq'
+version 'unspecified'
+
+apply plugin: 'java'
+apply plugin: 'application'
+
+sourceCompatibility = 1.8
+
+repositories {
+    mavenCentral()
+    jcenter()
+    
+    // Used for checking VANTIQ date format behavior with VANTIQ SDK
+    maven {
+    	url "https://dl.bintray.com/vantiq/maven"
+    }
+}
+configurations {
+    externalLibs
+}
+mainClassName = 'io.vantiq.extsrc.HikVisionSource.HikVisionMain'
+
+startScripts{
+    doLast{
+        def windowsScriptFile = file getWindowsScript()
+        def unixScriptFile = file getUnixScript()
+        // Add the log Config to the top of the classpath
+        windowsScriptFile.text = windowsScriptFile.text.replace("CLASSPATH=", "CLASSPATH=%APP_HOME%\\logConfig;")
+        unixScriptFile.text = unixScriptFile.text.replace('CLASSPATH=', 'CLASSPATH=$APP_HOME/logConfig:')
+    }
+}
+
+// Copies the logger setup into logConfig in the distribution
+applicationDistribution.from("src/main/resources") {
+    include "log4j2.xml"
+    into "logConfig"
+}
+
+tasks.withType(Test) {
+
+    if (rootProject.hasProperty("EntConTestSourceName")) {
+        systemProperty "EntConTestSourceName", rootProject.findProperty("EntConTestSourceName")
+    }
+    if (rootProject.hasProperty("TestAuthToken")) {
+        systemProperty "TestAuthToken", rootProject.findProperty("TestAuthToken") ?: "empty"
+    }
+    if (rootProject.hasProperty("TestVantiqServer")) {
+        systemProperty "TestVantiqServer", rootProject.findProperty("TestVantiqServer") ?: "empty"
+    }
+}
+if (System.env.HIKVISION_LOC == null ) {
+    throw new IllegalArgumentException("Required environment variable HIKVISION_LOC not found.")
+}
+
+dependencies {
+    // This dependency is used internally, and not exposed to consumers on their own compile classpath.
+    implementation 'com.google.guava:guava:23.0'
+    
+    compile project(':extjsdk')
+    
+    compile "org.slf4j:slf4j-api:1.7.25"
+    compile "org.apache.logging.log4j:log4j-slf4j-impl:2.11.0"
+    
+      
+    // Used for checking VANTIQ date format behavior with VANTIQ SDK
+    compile 'io.vantiq:vantiq-sdk:1.0.27'
+    
+    // Use JUnit test framework
+    testCompile 'junit:junit:4.12'
+    testCompile project(path:":extjsdk", configuration:"testArtifacts")
+
+    // Used to create a connection pool if asynchronous processing has been specified for publish/query handlers
+    compile group: 'com.zaxxer', name: 'HikariCP', version: '3.3.1'
+
+    // https://mvnrepository.com/artifact/net.java.dev.jna/jna
+    //compile group: 'net.java.dev.jna', name: 'jna', version: '5.6.0'
```

Comment:
Is there a reason you don't use the generally available jna.jar file?  You should unless there's a reason not to.

It would be
`compile group: 'net.java.dev.jna', name: 'Jan', version: '5.6.0'`

Don't need the https part.  Should be included from maven central, so you don't need to add that.

Reply:
the concern is that example.jar is complied with older version of jna.jar. 
the heep is  a left over i kept (temporarily) because i tried to download from maven and it didnt worked for me. with 5.6.0 version , but i can retry again ,

Reply:
please do.  That will simplify things.

Reply:
i tried , but the java project still add the ${System.env.HIKVISION_LOC}/jna.jar to the classpath , although i removed it . 
it still an open issue .